cmd/kueue/main.go

```diff
@@ -135,9 +136,11 @@ func main() {
 	if kubeConfig.UserAgent == "" {
 		kubeConfig.UserAgent = useragent.Default()
 	}
-	kubeConfig.QPS = *cfg.ClientConnection.QPS
-	kubeConfig.Burst = int(*cfg.ClientConnection.Burst)
-	setupLog.V(2).Info("K8S Client", "qps", kubeConfig.QPS, "burst", kubeConfig.Burst)
+
+	// Set the RateLimiter here, otherwise the controller-runtime's typedClient will use a different RateLimiter
+	// for each API type.
+	kubeConfig.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(*cfg.ClientConnection.QPS, int(*cfg.ClientConnection.Burst))
+	setupLog.V(2).Info("K8S Client", "qps", *cfg.ClientConnection.QPS, "burst", *cfg.ClientConnection.Burst)
```

Comment:
@alculquicondor @tenzen-y Do you think it can help us to create a config for the scope of the rate limiting settings (global vs per API type)?

Reply:
I would like to avoid adding configuration overhead.

This LGTM. I would just add an `ACTION REQUIRED` in the release note to adjust the qps.

Reply:
LGTM
One thing is documentation. Actually, we don't have docs on how to tune kueue-controller-managers client configurations, iiuc.

but, that is out of scope of this PR.